### PR TITLE
Add RealBox and RealVect

### PR DESCRIPTION
**Pull request by RTSandberg**
Add RealBox and RealVect to pyamrex, with tests

### CODE REVIEW

**Review by sayerhs — APPROVED** [review]

File: src/Base/RealBox.cpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        //             s <<  rb.hi(ii);
+        //             if (ii < AMREX_SPACEDIM - 1) {
+        //                 s << ", ";
+        //             } 
+        //         }
+        //         s << ">\n";
+        //         return s.str();
+        //     }
+        // )
+    // py::class_< IntVect >(m, "IntVect")
```
Comment (ax3l) [inline src/Base/RealBox.cpp]:
```suggestion

```

File: src/Base/RealBox.cpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        .def("intersects", &RealBox::intersects, "determine if box intersects with a box")
+        // .def("AlmostEqual")
+    ;
+    m.def("AlmostEqual", 
+            [](const RealBox& rb1, const RealBox& rb2, Real eps) {
+                return AlmostEqual(rb1,rb2,eps);
+            },
+            "Determine if two boxes are equal to within a tolerance",
+            py::arg("rb1"), py::arg("rb2"), py::arg("eps") = 0.0);
+}
```
Comment (ax3l) [inline src/Base/RealBox.cpp]:
```suggestion
}

```

File: src/Base/RealVect.cpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+#include <sstream>
+#include <optional>
+
+namespace py = pybind11;
+using namespace amrex;
+
+void init_RealVect(py::module &m) {
+
+
```
Comment (ax3l) [inline src/Base/RealVect.cpp]:
```suggestion
```

File: src/Base/RealBox.cpp [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+#include <AMReX_RealBox.H>
+
+#include <sstream>
+#include <optional>
+
+namespace py = pybind11;
+using namespace amrex;
+
+void init_RealBox(py::module &m) {
+    // py::class_< Direction >(m, "Direction");
```
Comment (ax3l) [inline src/Base/RealBox.cpp]:
```suggestion
```

File: src/Base/RealVect.cpp [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+          .def("__str",
+               [](const RealVect& rv) {
+                    std::stringstream s;
+                    s << rv;
+                    return s.str();
+               })
+
+          .def(py::init())
+#if (AMREX_SPACEDIM > 1)
+        .def(py::init<AMREX_D_DECL(Real, Real, Real)>())
```
Comment (ax3l) [inline src/Base/RealVect.cpp]:
```suggestion
          .def(py::init<AMREX_D_DECL(Real, Real, Real)>())
```

File: tests/test_realbox.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+    rb1 = amrex.RealBox([0,0,0], [1,2,1.5])
+    rb2 = amrex.RealBox([0,0,0], [1,2,1.5])
+    rb3 = amrex.RealBox([0,0.1,-0.1], [1,2,1.4])
+    assert(amrex.AlmostEqual(rb1,rb2))
+    assert(not amrex.AlmostEqual(rb1,rb3))
+    tol = 0.05
+    assert(not amrex.AlmostEqual(rb1,rb3,tol))
+    tol = 0.2
+    assert(amrex.AlmostEqual(rb1,rb3,tol))
```
Comment (ax3l) [inline tests/test_realbox.py]: Please end in a single newline at the end of files to make `git diff`s look cleaner (note the little sign it shows here on GitHub as well):
```suggestion
    assert(amrex.AlmostEqual(rb1,rb3,tol))

```

File: tests/test_realvect.py [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+                assert(ei[jj] == 0)
+
+
+
+
+
+
+
+
+
```
Comment (ax3l) [inline tests/test_realvect.py]:
```suggestion
```

**Review by ax3l — COMMENTED** [review]
Thanks a lot!

I marked a few locations inline but cannot yet identify the reason for the build issue on Windows.

**Comment by ax3l** [comment]
Ah, the windows issue came in independently in the last days, likely from an env or AMReX upstream update. Will try to fix separately.

**Comment by ax3l** [comment]
Windows fix in #33. When you update the PR with the suggestions tomorrow it will pass again :)

File: tests/test_realvect.py (hunk 7 above)
Comment (RTSandberg) [reply tests/test_realvect.py]: That was a lot of white space to eliminate!

File: src/Base/IntVect.cpp [hunk 8]
```diff
@@ -41,6 +41,7 @@ void init_IntVect(py::module &m) {
         .def(py::init<AMREX_D_DECL(int, int, int)>())
 #endif
         .def(py::init<int>())
+        .def(py::init<const std::array<int,AMREX_SPACEDIM>&>())
```
Comment (ax3l) [inline src/Base/IntVect.cpp:46]: Please include `<array>` and `<string>` in this file :)

File: src/Base/RealBox.cpp [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+#include <AMReX_Config.H>
+#include <AMReX_Array.H>
+#include <AMReX_Vector.H>
+#include <AMReX_REAL.H>
+#include <AMReX_IntVect.H>
+#include <AMReX_RealVect.H>
+#include <AMReX_Box.H>
+#include <AMReX_RealBox.H>
+
+#include <sstream>
```
Comment (ax3l) [inline src/Base/RealBox.cpp:20]:
```suggestion
#include <array>
#include <sstream>
#include <string>
```

File: src/Base/RealVect.cpp [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+#include <pybind11/pybind11.h>
+#include <pybind11/operators.h>
+#include <pybind11/stl.h>
+
+#include <AMReX_Config.H>
+#include <AMReX_RealBox.H>
+#include <AMReX_IntVect.H>
+
+#include <sstream>
+#include <optional>
```
Comment (ax3l) [inline src/Base/RealVect.cpp]:
```suggestion
#include <sstream>
#include <string>
#include <optional>
#include <vector>
```

File: tests/test_intvect.py [hunk 11]
```diff
@@ -115,3 +120,8 @@ def test_iv_conversions():
     obj = amrex.IntVect.max_vector().numpy()
     assert(isinstance(obj, np.ndarray))
     assert(obj.dtype == np.int32)
+
+    iv = amrex.IntVect(2)
```
Comment (ax3l) [inline tests/test_intvect.py:125]:
```suggestion
    # check that memory is not collected too early
    iv = amrex.IntVect(2)
```

File: src/Base/IntVect.cpp (hunk 8 above)
Comment (ax3l) [reply src/Base/IntVect.cpp:46]: done :)

**Review by ax3l — APPROVED** [review]
Thanks a lot, Ryan! :rocket: :sparkles: 

Congrats on your first PR to `pyAMReX`! :tada:

**Pull request merged**